go/vt/sqlparser/ast.go

```diff
@@ -2765,6 +2858,96 @@ type (
 )
 
 // iExpr ensures that only expressions nodes can be assigned to a Expr
+func (*Sum) iExpr()             {}
+func (*Min) iExpr()             {}
+func (*Max) iExpr()             {}
+func (*Avg) iExpr()             {}
+func (*CountStar) iExpr()       {}
+func (*Count) iExpr()           {}
+func (*GroupConcatExpr) iExpr() {}
+func (*BitAnd) iExpr()          {}
+func (*BitOr) iExpr()           {}
+func (*BitXor) iExpr()          {}
+func (*Std) iExpr()             {}
+func (*StdDev) iExpr()          {}
+func (*StdPop) iExpr()          {}
+func (*StdSamp) iExpr()         {}
+func (*VarPop) iExpr()          {}
+func (*VarSamp) iExpr()         {}
+func (*Variance) iExpr()        {}
+
+func (sum *Sum) GetArg() Expr                   { return sum.Arg }
```

Comment:
nit: this breaks the pattern this file had before. I think it's nice to keep all the iExpr() together. maybe we could move the `GetArg`, `IsDistinct` and `AggrName` to after the iExprs?